hw/snitch_cluster/src/snitch_cluster_wrapper.rdl.tpl

```diff
@@ -0,0 +1,44 @@
+// Copyright 2025 ETH Zurich and University of Bologna.
+// Solderpad Hardware License, Version 0.51, see LICENSE for details.
+// SPDX-License-Identifier: SHL-0.51
+
+${disclaimer}
+
+`ifndef __${cfg['cluster']['name'].upper()}_WRAPPER_RDL__
+`define __${cfg['cluster']['name'].upper()}_WRAPPER_RDL__
+
+`include "snitch_cluster_peripheral_reg.rdl"
+
+addrmap ${cfg['cluster']['name']}_wrapper #(
+    longint unsigned BASE_ADDR = ${hex(cfg['cluster']['cluster_base_addr'])}
+) {
+
+    default regwidth = ${cfg['cluster']['data_width']};
+
+    mem TCDM {
+        mementries = ${hex(int(cfg['cluster']['tcdm']['size'] * 1024 * 8 / cfg['cluster']['data_width']))};
+        memwidth = ${cfg['cluster']['data_width']};
+    };
+
+    mem BOOTROM {
+        mementries = ${hex(int(4 * 1024 * 8 / cfg['cluster']['data_width']))};
+        memwidth = ${cfg['cluster']['data_width']};
+    };
+
+    mem ZEROMEM {
+        mementries = ${hex(int(cfg['cluster']['zero_mem_size'] * 1024 * 8 / cfg['cluster']['data_width']))};
+        memwidth = ${cfg['cluster']['data_width']};
+    };
+
+
+    external TCDM                 TCDM           @BASE_ADDR;
```

Comment:
what does `external` mean in this context?

Reply:
`external` refers to the register being described external to the register block, i.e., requiring manual definition: https://peakrdl-regblock.readthedocs.io/en/latest/rdl_features/external.html#external-registers
From my understanding, it is similar to the `hwext` specification in the opentitan hjson format.